### PR TITLE
Adapter

### PR DESCRIPTION
**Pull request by ErykS1**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DriverAdapter.java [hunk 1]
```diff
@@ -4,14 +4,15 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DriverAdapter extends DrawPanelController implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DriverAdapter.java:12]: The name is better but it does not indicate the purpose of the adapter or what was actually adapted.

Inheritance is not used (no super drawLine calls). Should be removed!

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+
+public class LineDrawerAdapter extends LineFactory implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:10]: Inheriting from LineFactory is confusing.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Override
+    public void setPosition(int x, int y) {
+        this.startX = x;
+        this.startY = y;
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        ILine line = LineFactory.getSpecialLine();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:25]: Hardcoded line type (no OCP). What about 3.3.2 test?
Direct coupling with LineFactory class is unnecessary.

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java [hunk 4]
```diff
@@ -16,6 +16,11 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+		if(e.getActionCommand().equals("Figure Joe 1")) {
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:19]: Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.

**Review by kowallus — COMMENTED** [review]
2.5 pts 

Specific remarks are inlined in the code.
Comments about UML added to wikamp.

**Comment by ErykS1** [comment]
Zrobienie Lab2

**Pull request closed without merging**